src/http/endpoints/create_database.rs

```diff
@@ -20,6 +20,14 @@ async fn create_database(
     let entity_slug = &path.entity;
     let entity = ayb_db.get_entity_by_slug(entity_slug).await?;
     let db_type = get_header(&req, "db-type")?;
+
+    if entity.is_none() {
```

Comment:
I'm trying to understand the benefit of this change toward `Option`. Originally the code returned as an `AybError`, which felt appropriate because you were requesting a row and that row didn't exist. This is similar to other ORMs (e.g., [Django](https://docs.djangoproject.com/en/5.0/topics/db/queries/#retrieving-a-single-object-with-get)). This refactor instead introduces `Option`, but then all callers then return their own `AybError`, which would have happened in the original implementation with less boilerplate.

Just trying to weigh the benefits and costs of the refactor. Thank you for being patient with me! :)

Reply:
A caller might need to handle null values themselves (eg. user profiles may be null) and thus it cannot be mixed with _real_ errors like a TLS failure or a timeout.
Also, related to #103, different endpoints calling the same db function may need to return their own error code/error message instead of one determined by the database interface.

This PR is just a clean-up for future features (better errors, user profiles, etc).
I hope this explains the purpose if the PR :)

Reply:
I totally agree the caller *might* want to capture the case where a row doesn't exist. To your point, I think the solution is #103. Here's why I think we should consider moving to AybError with specific subtypes like RecordNotFound:
* It allows you to capture the error and handle it if you want to (you can choose to see if the return type is an `Err(RecordNotFound...)`)
* It allows you to not capture the error if you want to (meaning we wouldn't need the additional code in the call sites that this PR introduces), since you're not forced to pick between Some and None
* There's precedent in other ORMs (like Django, which is the only one I'm closely familiar with to be honest) to consider it an error
* It keeps the `list_...` and `get_...` behaviors more consistent in their return types, though only sort of (this is a weak argument)
* It moves us closer to #103, which is broadly good for the project (you've said this, so I'm not telling you anything new with this :))

I wonder if we can start toward #103 without requiring an codebase-wide rewrite. If we made AybError have just two variants for now:
```rust
enum AybError {
    RecordNotFound,
    DeprecatedAybError { message: String },
}
```

We could find and replace al instances of `AybError` construction with `AybError::DeprecatedAybError` construction for now, and only add the special variant as part of this PR. As a result, we could add all the special variants for other use cases in another PR while keeping this one small.

I know it's frustrating to have so much scrutiny on a small PR that is addressing a real issue, so I hope you'll have patience with me as we sort this out :).

Reply:
I think you're mostly right, I'll close this PR and implement the _improved_ AybError instead :)
That way I can implement the same idea as this PR and also partially fix #103. (Special offer! 2✕1 :p)

Reply:
Amazing! Thank you so much for doing this patiently! :)

Reply:
No problem! :)